### PR TITLE
Add new AuditLogAction types

### PR DESCRIPTION
**Pull request by bijij**
### Summary

Adds new audit log entry types specified in discordapp/discord-api-docs#1191
Also fixes an issue with unknown audit log types with extra data causing an exception.

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/audit_logs.py [hunk 1]
```diff
@@ -238,6 +238,16 @@ def _from_data(self, data):
                     'channel': self.guild.get_channel(channel_id) or Object(id=channel_id)
                 }
                 self.extra = type('_AuditLogProxy', (), elems)()
+            elif self.action.name.endswith('pin'):
+                # the pin actions have a dict wiht some information
+                channel_id = int(self.extra['channel_id'])
+                message_id = int(self.extra['message_id'])
+                elems = {
+                    'channel': self.guild.get_channel(channel_id) or Object(id=channel_id),
+                    'message_id': message_id
```
Comment (apple502j) [inline discord/audit_logs.py]: Possibly `self.guild._state._get_message()`?

File: discord/enums.py [hunk 2]
```diff
@@ -328,6 +337,9 @@ def category(self):
             AuditLogAction.unban:              None,
             AuditLogAction.member_update:      AuditLogActionCategory.update,
             AuditLogAction.member_role_update: AuditLogActionCategory.update,
+            AuditLogAction.member_move:        None,
+            AuditLogAction.member_move:        None,
```
Comment (apple502j) [inline discord/enums.py]: Duplicated line.

File: docs/api.rst [hunk 3]
```diff
@@ -1254,6 +1254,29 @@ of :class:`enum.Enum`.
 
         - :attr:`~AuditLogDiff.roles`
 
+    .. attribute:: member_move
+
+        A member's voice channel has beenn updated. This triggers when a
+        member is moved to a different voice channel.
+
+        When this is the action, the type of :attr:`~AuditLogEntry.target` is
```
Comment (apple502j) [inline docs/api.rst]: No: discordapp/discord-api-docs#1195

File: docs/api.rst [hunk 4]
```diff
@@ -1254,6 +1254,29 @@ of :class:`enum.Enum`.
...
+        the :class:`Member` or :class:`User` which was moved.
+
+    .. attribute:: member_disconnect
+
+        A member's voice state has changed. This triggers when a
+        member is force disconnected from voice.
+
+        When this is the action, the type of :attr:`~AuditLogEntry.target` is
```
Comment (apple502j) [inline docs/api.rst]: No again: discordapp/discord-api-docs#1195

File: discord/audit_logs.py [hunk 5]
```diff
@@ -238,6 +238,16 @@ def _from_data(self, data):
                     'channel': self.guild.get_channel(channel_id) or Object(id=channel_id)
                 }
                 self.extra = type('_AuditLogProxy', (), elems)()
+            elif self.action.name.endswith('pin'):
```
Comment (apple502j) [inline discord/audit_logs.py]: Move has some useful options - count and channel_id. Maybe do it here? Disconnect also has cont.

Comment (XuaTheGrate) [reply discord/audit_logs.py]: It seems like count never goes above one, so I'd say it's safe enough to leave out unless it changes in the future.

Comment (Rapptz) [reply discord/audit_logs.py]: count definitely goes above 1.

Comment (bijij) [reply discord/audit_logs.py]: My testing count does go above one yes, I'll have that in the extras now.

File: discord/enums.py (hunk 2 above)
Comment (bijij) [reply discord/enums.py]: Fixed, thanks.

File: discord/audit_logs.py [hunk 6]
```diff
@@ -238,16 +238,26 @@ def _from_data(self, data):
...
+                }
+                if self.action is enums.AuditLogAction.member_move:
+                    channel_id = int(self.extra['channel_id'])
+                    elems.update({
+                        'channel': self.guild.get_channel(channel_id) or Object(id=channel_id),
+                    })
+                self.extra = type('_AuditLogProxy', (), elems)()
             elif self.action.name.endswith('pin'):
-                # the pin actions have a dict wiht some information
+                # the pin actions have a dict wit some information
```
Comment (apple502j) [inline discord/audit_logs.py]:
```suggestion
                # the pin actions have a dict with some information
```

File: docs/api.rst [hunk 7]
```diff
@@ -1254,6 +1254,34 @@ of :class:`enum.Enum`.
 
         - :attr:`~AuditLogDiff.roles`
 
+    .. attribute:: member_move
+
+        A member's voice channel has beenn updated. This triggers when a
```
Comment (Modelmat) [inline docs/api.rst]:
```suggestion
        A member's voice channel has been updated. This triggers when a
```

File: docs/api.rst [hunk 8]
```diff
@@ -1254,6 +1254,34 @@ of :class:`enum.Enum`.
...
+        member is moved to a different voice channel.
+
+        When this is the action, the type of :attr:`~AuditLogEntry.extra` is
+        set to an unspecified proxy object with two attributes:
+
+        - ``channel``: A :class:`TextChannel` or :class:`Object` with the channel ID of which the members were moved to.
```
Comment (Modelmat) [inline docs/api.rst]: Perhaps?
```suggestion
        - ``channel``: A :class:`TextChannel` or :class:`Object` which corresponds to the channel where the member was moved.
```

Comment (Modelmat) [reply docs/api.rst]: Or actually, following the existing style:
```suggestion
        - ``channel``: A :class:`TextChannel` or :class:`Object` with the channel ID where the members were moved.
```

File: discord/audit_logs.py (hunk 1 above)
Comment (bijij) [reply discord/audit_logs.py]: Don't think so.
Audit logs aren't sent via WS events so there's no way to know when exactly to make a cache of the message.

**Comment by jack1142** [comment]
It's not in description of this PR but it resolves #2426

File: discord/enums.py [hunk 9]
```diff
@@ -341,6 +353,9 @@ def category(self):
             AuditLogAction.emoji_update:       AuditLogActionCategory.update,
             AuditLogAction.emoji_delete:       AuditLogActionCategory.delete,
             AuditLogAction.message_delete:     AuditLogActionCategory.delete,
+            AuditLogAction.integration_create: AuditLogActionCategory.create,
+            AuditLogAction.integration_update: AuditLogActionCategoty.update,
```
Comment (Guy293) [inline discord/enums.py]: You have a mistype here,
AuditLogActionCategoty
Instead of
AuditLogActionCategory

Comment (Rapptz) [reply discord/enums.py]: This was already fixed.

Comment (Harmon758) [reply discord/enums.py]: See https://github.com/Rapptz/discord.py/commit/f9b8b234e314cd182c39ec0a55f466d81b9926c7.

**Pull request closed without merging**